configuration/pih/scripts/global/zl.js

```diff
@@ -0,0 +1,144 @@
+function setUpEdd(currentEncounterDate, msgWeeks) {
+  var updateEdd = function () {
+    var lastPeriodDate = htmlForm.getValueIfLegal("lastPeriodDate.value");
+    if (
+      typeof lastPeriodDate !== "undefined" &&
+      lastPeriodDate !== null &&
+      lastPeriodDate.length > 0
+    ) {
+      var today = new Date();
+      if (
+        typeof currentEncounterDate !== "undefined" &&
+        currentEncounterDate !== null &&
+        currentEncounterDate.length > 0
+      ) {
+        // calculate the gestational age at the time of the encounter
+        today = new Date(+currentEncounterDate);
+      }
+
+      var dateObj = getField("lastPeriodDate.value").datepicker("getDate");
+      var newDate = new Date(dateObj);
+      // time difference
+      var timeDiff = Math.abs(today.getTime() - newDate.getTime());
+      // weeks difference = gestational age
+      var diffWeeks = Math.ceil(timeDiff / (1000 * 3600 * 24 * 7));
+
+      // Estimated Delivery Date = (LMP - 3 months) + 12 months + 7 days
+      newDate.setMonth(newDate.getMonth() - 3);
+      newDate.setFullYear(newDate.getFullYear() + 1);
+      newDate.setDate(newDate.getDate() + 7);
```

Comment:
![Screenshot from 2021-07-30 12-08-31](https://user-images.githubusercontent.com/1031876/127700650-dd7ea9e3-ab67-4baa-8354-10dd9cfdfa9d.png)

Setting a negative month does not decrement the year, so this will result in 21-month pregnancies for women having their last period before April.

Reply:
To begin, create EDD as its own date, rather than mutating the `lastPeriodDate`:

```js
const edd = new Date(lastPeriodDate);
```

Then do the math better. Actually you could do it in one line:

```js
const edd = new Date(lastPeriodDate.getTime() + 1000 * 60 * 60 * 24 * 280);
```

Reply:
Oh shoot, I'm sorry, I just realized that this code wasn't part of this PR, I just moved it! My bad. Anyway who knows how long the EDD calculation has been wrong for 1/4 of patients.

Reply:
Thanks @brandones for noticing this!  @lnball we probably should ticket this as a separate ticket, mainly to make sure we test it fully after the changes?

Reply:
https://pihemr.atlassian.net/browse/UHM-5676